packages/drivers/debugger/src/sanitize.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This tool cleans up a message.json file downloaded through fluid-fetch to remove
+ * user content and user identifying information.  Enough information can be retained
+ * to allow loading through Fluid Preview, or everything can be scrubbed so that only
+ * replay-tool can read the result.  Anonymous identifying information such as client
+ * IDs are always retained.  Object keys are NOT scrubbed, including those that are
+ * nested within values (only leaf values are scrubbed).
+ *
+ * Note: While user content/information is scrubbed, it should not be assumed to be
+ * fully anonymized because certain meta-information (such as word lengths and
+ * consistent replacement) are preserved.
+ *
+ * Messages must match known structures when scrubbing for Fluid Preview.
+ */
+
+import fs from "fs";
+import process from "process";
+import {
+    ISequencedDocumentMessage,
+} from "@fluidframework/protocol-definitions";
+import { Sanitizer } from "./sanitizer";
+
+function printUsage() {
+    console.log("Usage:");
+    console.log("   node sanitize [--full | --noBail] <input>");
+    console.log("Where");
+    console.log("  [--full] - scrub fully (result cannot be loaded in Fluid Preview)");
+    console.log("  [--noBail] - don't bail out when encountering an unknown message format (it won't be scrubbed");
+    console.log("  <input> - file path to message.json - file downloaded by FluidFetch tool");
+    console.log("Note: <input> is sanitized in place");
+    process.exit(-1);
+}
+
+function Sanitize(msgPath: string, fullScrub: boolean, noBail: boolean) {
+    const input = fs.readFileSync(msgPath, { encoding: "utf-8" });
```

Comment:
Please note that fetch tool will write out messages.json, then messages01.json, messages02.json and so on for big files. Is this handled here? Should we instead use file driver that I believe already handles that? Or some other abstraction not to spread this knowledge all over code base?

Reply:
It's not handled here, and it wasn't handled before this refactoring either.  I've also never seen the splitting; all the testing I dd was on the readout document which is probably one of the largest docs out there and it gave me a single file.  Given it isn't new behavior I'll open an issue to track that separately.